TYPO3.Neos/Documentation/CreatingASite/ContentCache.rst

```diff
@@ -182,6 +183,19 @@ The ``ContentCollection`` cache configuration declares a tag that will flush the
 any of it's descendants (direct or indirect child) changes. So editing a node inside the collection will flush the
 whole collection cache entry and cause it to re-render.
 
+.. note::
+  When using ``cached`` as the cache mode, your ``entryTags`` should always contain the node identifier. Otherwise, the
+  cache will not be flushed when you make changes to the node itself, which will lead to unexpected behavior in the Neos
```

Comment:
It's not only the rendering output, it's the Neos backend also. Try creating a NodeType with a custom property of type reference, and set mode "cached" without adding the node identifier to the entry tags. 
In the backend, that lead to me e.g. setting the reference to some other page, but after the UI refreshed the link was gone again in the inspector.